### PR TITLE
Fix tagging ISO structures

### LINKED ISSUE #116: Analyze usage of Tag 24 in ISO structures
Picking up the work of #114, we should analyze the usage of CBOR tags `24` in our library, and read through the mDL spec. Maybe it's enough to annotate `ByteStringWrapper` with the tag in general.

### PR DESCRIPTION
**Pull request by nodh**
Resolves #116

### CODE REVIEW

**Review by JesusMcCloud — COMMENTED** [review]
everything else looks very clean!

**Comment by JesusMcCloud** [comment]
obviously something's up, with one test failing.

File: vck/src/commonMain/kotlin/at/asitplus/wallet/lib/agent/Validator.kt [hunk 1]
```diff
@@ -304,12 +304,10 @@ class Validator(
     }
 
     private fun ByteStringWrapper<IssuerSignedItem>.verify(mdlItems: ValueDigestList?): Boolean {
-        val issuerHash = mdlItems?.entries?.firstOrNull { it.key == value.digestId } ?: return false
-        // TODO analyze usages of tag wrapping
+        val issuerHash = mdlItems?.entries?.firstOrNull { it.key == value.digestId }
+            ?: return false
         val verifierHash = serialized.wrapInCborTag(24).sha256()
```
[the remark this answers is not in the document]
Comment (nodh) [reply vck/src/commonMain/kotlin/at/asitplus/wallet/lib/agent/Validator.kt:314]: Unfortunately not, see my comment in 3ab0f8f1

**Review by JesusMcCloud — APPROVED** [review]
looks good! pending tests

**Pull request merged**